packages/spindle-ui/src/InlineNotification/InlineNotification.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {
+  ReactNode,
+  FC,
+  Children,
+  cloneElement,
+  ReactElement,
+  HTMLAttributes,
+} from 'react';
+import { Button as SpindleButton } from '../Button/Button';
+import { TextButton as SpindleTextButton } from '../TextButton/TextButton';
+import { IconButton as SpindleIconButton } from '../IconButton/IconButton';
+
+type Variant = 'information' | 'confirm' | 'error';
+type Emphasis = 'normal' | 'emphasized';
+type Layout = 'inset' | 'full';
+
+type Props = {
+  children?: ReactElement;
+  variant?: Variant;
+  emphasis?: Emphasis;
+  layout?: Layout;
+  visible?: boolean;
+};
+
+const BLOCK_NAME = 'spui-InlineNotification';
+const DEFAULT_VARIANT = 'information';
+const DEFAULT_EMPHASIS = 'normal';
+const DEFAULT_LAYOUT = 'inset';
+
+type ButtonVariant = 'neutral' | 'danger' | 'contained' | 'outlined';
+const computedButtonVariant = (
+  variant: Variant,
+  emphasis: Emphasis,
+): ButtonVariant => {
+  if (variant === 'confirm') {
+    if (emphasis === 'normal') {
+      return 'outlined';
+    } else {
+      return 'contained';
+    }
+  } else if (variant === 'error') {
+    return 'danger';
+  } else {
+    return 'neutral';
+  }
+};
+
+const Frame: FC<Props> = ({
+  children,
+  variant = DEFAULT_VARIANT,
+  emphasis = DEFAULT_EMPHASIS,
+  layout = DEFAULT_LAYOUT,
+  visible = false,
+}) => {
+  return (
+    <div
+      className={`${BLOCK_NAME}--${layout} ${BLOCK_NAME}--${variant}${emphasis}`}
+      hidden={!visible}
+    >
+      <div className={`${BLOCK_NAME}-content`}>
+        {Children.map(children, (child) =>
+          child ? cloneElement(child, { variant, emphasis }) : child,
+        )}
+      </div>
+    </div>
+  );
+};
+
+const Icon: FC<{
+  children: ReactNode;
+  variant: Variant;
+  emphasis: Emphasis;
+}> = ({ children, variant, emphasis }) => (
+  <div
+    className={`${BLOCK_NAME}-left ${BLOCK_NAME}-icon ${BLOCK_NAME}-icon--${variant}${emphasis}`}
+  >
+    {children}
+  </div>
+);
+const Avatar: FC<{ src: string; alt: string }> = ({ src, alt }) => (
+  <img
+    className={`${BLOCK_NAME}-left ${BLOCK_NAME}-avatar`}
+    src={src}
+    alt={alt}
+  />
+);
+const Text: FC<{
+  children: ReactNode;
+  variant: Variant;
+  emphasis: Emphasis;
+}> = ({ children, variant, emphasis }) => (
+  <span
```

Comment:
ﾋﾟｰの方が :yosagesan: かな？ 🤔 
```suggestion
  <p
```